### ISSUE TITLE
Create Commitment: Set commitment period

### ISSUE THREAD

**Issue by slgraham** [CommitPool/CommitPool-monorepo#31]
**User Story / Job to be Done / Problem**
- As a user creating a new commitment, I want to customize my commitment period so that it fits into my life and goals.
- As a user creating a new commitment, I want to know exactly what day/time my commitment is set to start and end so that I can schedule my activity appropriately.

**Potential solution ideas**
- date pickers
- Picking today's date to start would set the start time to now, and picking any other day would set the star time to 0:00 for that day. And end time is always 23:59 for the end date selected

**Comment by bitbeckers** [comment]
"As a user creating a new commitment, I want to customize my commitment period so that it fits into my life and goals."

@ryoungblom mentioned that instead of a datepicker setting amount of days (like in the current app) might make more sense. What did you have in mine when drafting this? Because I can see the value of not letting user do some sort of math and just say "start in 2 days lasting for 5 days"

**Comment by ryoungblom** [comment]
Yes, my thought was that most people think about setting goals as "I want to do X within Y number of Z"  (e.g., I want to do X within 7 number of days" or "I want to do X within 3 number of months"), as opposed to "I want to do X between Y date and Z date."

I suspect that most users will want to set a start date and then give themselves some number of days or weeks to finish, rather than setting a start date and then figuring out the end date.  Like, I think about myself, and how long it would take me to say, "ok, today is the 25th of May, so 2 weeks from now... that's 14 days... how many days are in May?  So I guess... uh..." and it would be one million times easier to just select "2 weeks."

That said—there may be the odd person who wants to have a specific date!  Maybe we have the day/week/etc selector as the default but a button to switch to a date picker?  Some people might want to set goals before, say, a birthday or a wedding or something, and we don't want to force them to do all that math converting dates, either.

@bitbeckers why can't you guys come up with a metric version of dates, where each day is 10 hours long and there are 10 days in a week and 10 weeks in a month and 10 months in a year??  C'mon.

**Comment by bitbeckers** [comment]
This helped me to believe it might make more sense to think in amount of days. The population of people using commitpool for a wedding is probably way smaller than all users and in that case, just do the math. We might learn from users that they'd prefer a datepicker, but for now we could use the amount of days.

**Comment by slgraham** [comment]
Let's go with what you guys are suggesting. It's a very cheap decision in that it'll be very easy to change if we discover its not right.

**Comment by ryoungblom** [comment]
> Let's go with what you guys are suggesting. It's a very cheap decision in that it'll be very easy to change if we discover its not right.

What's the suggestion?  I think that both a pure days interface and a pure days interface with the option to switch to a date picker were discusses, and there's very little technical overhead for either.... it's purely a UI decision at this point

**Comment by slgraham** [comment]
> there's very little technical overhead for either.... it's purely a UI decision at this point
exactly, so I don't think it matters much which we pick now. So let's go with the pure days interface (no calendar/date picker).

What is important is displaying the dates selected as the user changes the days.

**Comment by bitbeckers** [comment]
Which can be done easily. So we're going with amount of days as input with a from-to field displaying the date range?
